File: Examen_Api/Models/Pago.cs
using System.ComponentModel.DataAnnotations;

namespace Examen_Api.Models
{
    public class Pago
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Placa { get; set; }

        [Required]
        public string NombrePeaje { get; set; }

        [Required]
        [RegularExpression(@"^[IVXVI]+$|^I{1,5}$")]
        public string IdCategoriaTarifa { get; set; }

        [Required]
        public string FechaRegistro { get; set; }

        [Required]
        [Range(0, int.MaxValue)]
        public int Valor {  get; set; }
    }
}

File: Flutter/models/pago.dart
class Pago {
  int id;
  String placa;
  String nombrePeaje;
  String idCategoriaTarifa;
  String fechaRegistro;
  int valor;

  Pago({
    required this.id,
    required this.placa,
    required this.nombrePeaje,
    required this.idCategoriaTarifa,
    required this.fechaRegistro,
    required this.valor,
  });

  factory Pago.fromJson(Map<String, dynamic> json) => Pago(
    id: json['id'],
    placa: json['placa'],
    nombrePeaje: json['nombrePeaje'],
    idCategoriaTarifa: json['idCategoriaTarifa'],
    fechaRegistro: json['fechaRegistro'],
    valor: json['valor'],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'placa': placa,
    'nombrePeaje': nombrePeaje,
    'idCategoriaTarifa': idCategoriaTarifa,
    'fechaRegistro': fechaRegistro,
    'valor': valor,
  };
}